homeassistant/components/aurora_abb_powerone/aurora_device.py

```diff
@@ -0,0 +1,53 @@
+"""Top level class for AuroraABBPowerOneSolarPV inverters and sensors."""
+import logging
+
+from aurorapy.client import AuroraSerialClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ATTR_DEVICE_NAME,
+    ATTR_FIRMWARE,
+    ATTR_MODEL,
+    ATTR_SERIAL_NUMBER,
+    DEFAULT_DEVICE_NAME,
+    DOMAIN,
+    MANUFACTURER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AuroraDevice(Entity):
+    """Representation of an Aurora ABB PowerOne device."""
+
+    def __init__(self, client: AuroraSerialClient, config_entry: ConfigEntry):
+        """Initialise the basic device."""
+        self.config_entry = config_entry
+        self.type = "device"
+        self.client = client
+        self._available = True
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique id for this device."""
+        serial = self.config_entry.data.get(ATTR_SERIAL_NUMBER, "dummy sn")
```

Comment:
https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

If the serial number if optional we should use the config entry id https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-of-last-resort

Reply:
If ATTR_SERIAL_NUMBER will always be defined then use `self.config_entry.data[ATTR_SERIAL_NUMBER]`

Reply:
Ok. Done.  Thanks for the review! :)